src/io/flutter/survey/FlutterSurveyNotifications.java

```diff
@@ -94,26 +72,23 @@ private void check(@NotNull VirtualFile file) {
     });
   }
 
-  @NotNull final Project myProject;
-
-  FlutterSurveyNotifications(@NotNull Project project) {
-    this.myProject = project;
-  }
-
   private void checkForDisplaySurvey() {
```

Comment:
@pq - it looks like the following code happens synchronously on whatever thread we were called on (potentially the EDT)?

You may change this to working on a background thread (and if necessary, switching back the EDT at the very end).

We use a few different ways to execute in background threads - one is `ApplicationManager.getApplication().executeOnPooledThread()` (not sure if other's may be more appropriate).

Reply:
Yup. I'm looking at this very thing! Sorry I missed it first time through!